ui/src/app/login/page.tsx

```diff
@@ -1,14 +1,26 @@
 "use client"
 
-import { login } from "../queries"
-import { useMutation } from "react-query"
+import { getStatus, login } from "../queries"
+import { useMutation, useQuery } from "react-query"
 import { useState, ChangeEvent } from "react"
 import { useCookies } from "react-cookie"
 import { useRouter } from "next/navigation"
+import { useAuth } from "../auth/authContext"
+
+type statusResponse = {
+    initialized: boolean
+}
 
 export default function LoginPage() {
     const router = useRouter()
+    const auth = useAuth()
     const [cookies, setCookie, removeCookie] = useCookies(['user_token']);
+    const statusQuery = useQuery<statusResponse, Error>({
+        queryFn: () => getStatus()
+    })
+    if (!auth.firstUserCreated && (statusQuery.data && !statusQuery.data.initialized)) {
+        router.push("/onboarding")
```

Comment:
```suggestion
        router.push("/initialize")
```

It may be a personal preference, but `initialize` seems more appropriate for what we are trying to do here.

Reply:
changed